### DISCUSSION TITLE
mongodb query 动态数据提示不是JSON格式数据的问题

### DISCUSSION THREAD

**Discussion by 842797524** [illacloud/illa-builder#3593] [Q&A]
![image](https://github.com/illacloud/illa-builder/assets/33978392/2bd6fcdd-6fa2-47f4-aa76-d885c1b01818)
![image](https://github.com/illacloud/illa-builder/assets/33978392/bb9da5f6-76db-4b41-8d8c-23bd4b572f09)
自部署，使用动态语法不能通过检查，是我哪里做错了吗？希望可以提供教程地址或者实例，感谢！

**Comment by owenlongbo** [comment]
"{{ transformer1.value }}"

**Reply by 842797524** [reply]
对于string格式这种是适用的，但是如果是上面array格式的数据会报错。
![image](https://github.com/illacloud/illa-builder/assets/33978392/9e64ebe0-11c6-4095-b34f-a63e131040aa)
![image](https://github.com/illacloud/illa-builder/assets/33978392/19b4c2ec-6474-4dd0-8e63-004ac71f39b7)

**Reply by owenlongbo** [reply]
这个问题是这样的，我们没有做动态类型转换，所以您需要在string拼接的时候使用JSON.stringify转换一下
{
  "category": {"$in": {{ JSON.stringify(transformer1.value) }} }
}
